src/components/Chip/Chip.tsx

```diff
@@ -89,27 +92,49 @@ const Chip = ({
     </>
   );
 
-  const chipClassName = classNames({
-    [`p-chip--${appearance}`]: !!appearance,
-    "p-chip": !appearance,
-  });
+  const chipClassName = classNames(
+    {
+      [`p-chip--${appearance}`]: !!appearance,
+      "p-chip": !appearance,
+    },
+    props.className
+  );
+
+  const commonProps = {
+    ...props,
+    "aria-pressed": selected,
+    className: chipClassName,
+    onClick: onClick,
+  };
 
   if (onDismiss) {
     return (
-      <span className={chipClassName} aria-pressed={selected} {...props}>
+      <span
+        {...commonProps}
+        onKeyDown={(e) => onKeyDown<HTMLSpanElement>(e)}
+        role="button"
```

Comment:
Buttons can't be nested in each other. That's why chip with dismiss button is not interactive itself. It's bad from UX and accessibility point of view.

What kind of interactions do you need on dismissable chip other than dismissing it?

Reply:
I've removed this part from the PR following the decision in https://github.com/canonical-web-and-design/vanilla-framework/issues/4366